### PR TITLE
Adds the TAT-36 AT Gun

### PR DESCRIPTION
**Pull request by Pariah919**
## About The Pull Request
https://streamable.com/z5iobq
Adds the TAT-36 AT Gun, a delpoyed and moveable weapon that is meant to be used by mutliple people to be used effectively.

It has 3 types of shells, one solid shot that pierces walls and mobs alike, and two different kinds of explosives, one pens walls before exploding, but has lesser charge, and the last one is a large AOE HE.

Aim mode delay of 3 seconds.

Reload delay of a second.

It has a zoom delay of 4 seconds.

It holds ten rounds in its sponson, and can be pulled out by right clicking.

It is moveable once deployed, but cannot be undeployed (may be changed to spawn deployed or in a box)

It takes 4 seconds for it to enter aim mode! I added a var for this to guns.

Sprites by Fitz Sholl, credit to him. Based gun.

## Why It's Good For The Game
Gives a unique weapon to marines we haven't seen before.

## Changelog
:cl:
add: TAT-36 AT gun, a deployed anti tank gun, with several shell types.
code: adds a scope time zoom stat, only the TAT-36 uses it, which takes 3 seconds to enter scoped status. Also adds an aim time speed for time that it takes to enter aim mode.
/:cl:

### CODE REVIEW

**Comment by Moggram** [comment]
The sprites need some work, but it would be cool to have a WW1 field gun.

**Comment by Pariah919** [comment]
>The sprites need some work, but it would be cool to have a WW1 field gun.
>
they're entirely placeholder rn so i could test it

**Comment by Lefunny** [comment]
we already know by xray weapons with can ignore walls and even xenos, this weapon sounds real powercreep, also how much for it?

**Comment by InterroLouis** [comment]
Pfffft that undeploy time.

File: code/modules/projectiles/magazines/mounted.dm [hunk 1]
```diff
@@ -40,3 +40,25 @@
 	max_rounds = 100
 	default_ammo = /datum/ammo/bullet/rifle/heavy
 	reload_delay = 1 SECONDS
+
+/obj/item/ammo_magazine/standard_atgun
+	name = "TAT-36 AP-HE shell(37MM Shell)"
```
Comment (XSlayer300) [inline code/modules/projectiles/magazines/mounted.dm:45]:
```suggestion
	name = "TAT-36 AP-HE shell (37mm Shell)"
```

File: code/modules/projectiles/magazines/mounted.dm [hunk 2]
```diff
@@ -40,3 +40,25 @@
...
+	icon = 'icons/Marine/marine-atgun.dmi'
+	icon_state = "pak_ammo"
+	flags_magazine = MAGAZINE_REFUND_IN_CHAMBER
+	caliber = CALIBER_37MM
+	max_rounds = 1
+	default_ammo = /datum/ammo/rocket/atgun_shell
+	reload_delay = 1 SECONDS
+
+/obj/item/ammo_magazine/standard_atgun/apcr
+	name = "TAT-36 APCR shell (37MM Shell)"
```
Comment (XSlayer300) [inline code/modules/projectiles/magazines/mounted.dm:57]:
```suggestion
	name = "TAT-36 APCR shell (37mm Shell)"
```

File: code/modules/projectiles/magazines/mounted.dm [hunk 3]
```diff
@@ -40,3 +40,25 @@
...
+	desc = "A 37mm tungsten shell for light anti tank guns made to penetrate threw just about everything, but it won't leave a big hole."
+	default_ammo = /datum/ammo/rocket/atgun_shell/apcr
+
+/obj/item/ammo_magazine/standard_atgun/he
+	name = "TAT-36 HE (37MM Shell)"
```
Comment (XSlayer300) [inline code/modules/projectiles/magazines/mounted.dm:62]:
```suggestion
	name = "TAT-36 HE (37MM dhell)"
```

File: code/modules/projectiles/magazines/mounted.dm [hunk 4]
```diff
@@ -40,3 +40,25 @@
...
+	icon_state = "pak_ammo"
+	flags_magazine = MAGAZINE_REFUND_IN_CHAMBER
+	caliber = CALIBER_37MM
+	max_rounds = 1
+	default_ammo = /datum/ammo/rocket/atgun_shell
+	reload_delay = 1 SECONDS
+
+/obj/item/ammo_magazine/standard_atgun/apcr
+	name = "TAT-36 APCR shell (37MM Shell)"
+	desc = "A 37mm tungsten shell for light anti tank guns made to penetrate threw just about everything, but it won't leave a big hole."
```
Comment (XSlayer300) [inline code/modules/projectiles/magazines/mounted.dm:58]:
```suggestion
	desc = "A 37mm tungsten shell for light anti tank guns made to penetrate through just about everything, but it won't leave a big hole."
```
T H R E W

File: code/modules/projectiles/magazines/mounted.dm [hunk 5]
```diff
@@ -40,3 +40,25 @@
...
+	default_ammo = /datum/ammo/rocket/atgun_shell/apcr
+
+/obj/item/ammo_magazine/standard_atgun/he
+	name = "TAT-36 HE (37MM Shell)"
+	desc = "A 37mm shell for light anti tank guns made to destroy fortifications, the high amount of payload gives it a slow speed. But it leave quite a hole."
```
Comment (XSlayer300) [inline code/modules/projectiles/magazines/mounted.dm:63]:
```suggestion
	desc = "A 37mm shell for light anti tank guns made to destroy fortifications, the high amount of payload gives it a slow speed. But it leaves quite a hole."
```

**Comment by novaepee** [comment]
> we already know by xray weapons with can ignore walls and even xenos, this weapon sounds real powercreep, also how much for it?

X-ray weapon is actually not strong.

**Comment by Pariah919** [comment]
This is pretty much done, save for longterm balancing. I suppose.

**Comment by yyzsong** [comment]
>Cooperation weapon

either it will not be used at all or some metagang will hop on and absolutely destroy xenos with it
10/10 great PR

**Comment by Lefunny** [comment]
> 

Yeah this one will be a xray weapon with heavy damage and funny shells

File: code/modules/projectiles/guns/mounted.dm [hunk 6]
```diff
@@ -201,3 +201,89 @@
...
+	scatter = 0
+	recoil = 3
+	fire_delay = 3 SECONDS
+	burst_amount = 1
+	deploy_time = 1 SECONDS
+	undeploy_time = 2000 SECONDS
+	max_integrity = 2000
+
+/obj/item/weapon/gun/standard_atgun/Initialize()
+	AddElement(/datum/element/deployable_item, /obj/machinery/deployable/mounted/atgun, 5 SECONDS)
```
Comment (hyper2snyper) [inline code/modules/projectiles/guns/mounted.dm:247]: yeah this is bad. Take /gun init and make the element addition there based on a gun var.

File: code/modules/projectiles/guns/mounted.dm [hunk 7]
```diff
@@ -201,3 +201,89 @@
...
+	return sponson.open(user)
+
+obj/item/storage/internal/ammo_rack/handle_mousedrop(mob/user, obj/over_object)
+	if(!ishuman(user))
+		return FALSE
+
+	if(user.lying_angle || user.incapacitated()) //Can't use your inventory when lying
+		return FALSE
+
+	if(istype(user.loc, /obj/vehicle/multitile/root/cm_armored)) //Stops inventory actions in a mech/tank
```
Comment (hyper2snyper) [inline code/modules/projectiles/guns/mounted.dm:274]: this and the two above it should be one check

**Review by hyper2snyper — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/guns/mounted.dm (hunk 6 above)
Comment (Pariah919) [reply code/modules/projectiles/guns/mounted.dm]: done

File: code/__DEFINES/calibers.dm [hunk 8]
```diff
@@ -59,6 +59,8 @@
 #define CALIBER_84MM "84mm rocket"
 #define CALIBER_67MM "67mm shell"
 #define CALIBER_68MM "68mm rocket"
+#define CALIBER_37MM "37mm" //Light AT Cannon
+
```
Comment (TiviPlus) [inline code/__DEFINES/calibers.dm:63]:
```suggestion
#define CALIBER_37MM "37mm"

```

File: code/modules/projectiles/gun_attachables.dm [hunk 9]
```diff
@@ -668,6 +670,14 @@ inaccurate. Don't worry if force is ever negative, it won't runtime.
...
 	zoom_tile_offset = 3
 
+/obj/item/attachable/scope/unremovable/standard_atgun
+	name = "TAT-36 long range scope"
+	desc = "An unremovable set of long range scopes, very complex to properly range. Requires time to aim.."
+	icon_state = "sniperscope_invisible"
+	flags_attach_features = ATTACH_ACTIVATION
+	scope_delay = 4 SECONDS
+
+
```
Comment (TiviPlus) [inline code/modules/projectiles/gun_attachables.dm:680]:
```suggestion
/obj/item/attachable/scope/unremovable/standard_atgun
	name = "TAT-36 long range scope"
	desc = "An unremovable set of long range scopes, very complex to properly range. Requires time to aim.."
	icon_state = "sniperscope_invisible"
	flags_attach_features = ATTACH_ACTIVATION
	scope_delay = 4 SECONDS
```

File: code/modules/projectiles/gun_attachables.dm [hunk 10]
```diff
@@ -1409,15 +1421,15 @@ inaccurate. Don't worry if force is ever negative, it won't runtime.
 	if(!isgun(attaching_to))
 		return FALSE
 	var/obj/item/weapon/gun/attaching_gun = attaching_to
-	if(CHECK_BITFIELD(attaching_gun.flags_gun_features, GUN_IS_SENTRY))
+	if(istype(master_gun, /obj/item/weapon/gun/sentry))
 		to_chat(attacher, span_warning("[attaching_gun] is already a sentry!"))
```
Comment (TiviPlus) [inline code/modules/projectiles/gun_attachables.dm:1425]: why are you removing flags

File: code/modules/projectiles/gun_system.dm [hunk 11]
```diff
@@ -336,11 +340,9 @@
...
 	muzzle_flash = new(src, muzzleflash_iconstate)
 
-	if(flags_item & IS_DEPLOYABLE)
-		if(flags_gun_features & GUN_IS_SENTRY)
-			AddElement(/datum/element/deployable_item, /obj/machinery/deployable/mounted/sentry, deploy_time, undeploy_time)
-		else
-			AddElement(/datum/element/deployable_item, /obj/machinery/deployable/mounted, deploy_time, undeploy_time)
+	if(flags_item)
+		if(deployed_item)
+			AddElement(/datum/element/deployable_item, deployed_item, deploy_time, undeploy_time)
```
Comment (TiviPlus) [inline code/modules/projectiles/gun_system.dm:345]: ????

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Review by Kuro020 — APPROVED** [review]

File: code/modules/projectiles/gun_attachables.dm (hunk 10 above)
Comment (Pariah919) [reply code/modules/projectiles/gun_attachables.dm:1425]: hyper told me to do this and the one below

**Review by Pariah919 — COMMENTED** [review]
whoops

File: code/modules/projectiles/gun_attachables.dm [hunk 12]
```diff
@@ -1409,15 +1419,15 @@ inaccurate. Don't worry if force is ever negative, it won't runtime.
 	if(!isgun(attaching_to))
 		return FALSE
 	var/obj/item/weapon/gun/attaching_gun = attaching_to
-	if(CHECK_BITFIELD(attaching_gun.flags_gun_features, GUN_IS_SENTRY))
+	if(istype(master_gun, /obj/item/weapon/gun/sentry))
```
Comment (hyper2snyper) [inline code/modules/projectiles/gun_attachables.dm:1422]:
```suggestion
	if(ispath(attaching_gun.deploy_type, /obj/machinery/deployable/mounted/sentry))
```

File: code/modules/projectiles/gun_system.dm [hunk 13]
```diff
@@ -336,11 +340,9 @@
 
 	muzzle_flash = new(src, muzzleflash_iconstate)
 
-	if(flags_item & IS_DEPLOYABLE)
-		if(flags_gun_features & GUN_IS_SENTRY)
-			AddElement(/datum/element/deployable_item, /obj/machinery/deployable/mounted/sentry, deploy_time, undeploy_time)
-		else
-			AddElement(/datum/element/deployable_item, /obj/machinery/deployable/mounted, deploy_time, undeploy_time)
+	if(flags_item)
```
Comment (hyper2snyper) [inline code/modules/projectiles/gun_system.dm:343]: delete this,
you only need the `if(deployed_item)` check

i told you this already

File: code/modules/projectiles/guns/mounted.dm [hunk 14]
```diff
@@ -201,3 +204,82 @@
...
+
+/obj/machinery/deployable/mounted/atgun/attack_hand_alternate(mob/living/user)
+	return sponson.open(user)
+
+obj/item/storage/internal/ammo_rack/handle_mousedrop(mob/user, obj/over_object)
+	if(!ishuman(user))
+		return FALSE
+
+	if(user.lying_angle || user.incapacitated()) //Can't use your inventory when lying
+		return FALSE
```
Comment (hyper2snyper) [inline code/modules/projectiles/guns/mounted.dm:271]: merge this with the one above

File: code/modules/projectiles/guns/mounted.dm [hunk 15]
```diff
@@ -201,3 +204,82 @@
...
+
+	if(over_object == user && Adjacent(user)) //This must come before the screen objects only block
+		open(user)
+		return FALSE
+
+/obj/machinery/deployable/mounted/atgun/ex_act(severity)
```
Comment (hyper2snyper) [inline code/modules/projectiles/guns/mounted.dm:277]: this take different amounts of explosion damage?

File: code/modules/projectiles/guns/mounted.dm [hunk 16]
```diff
@@ -201,3 +204,82 @@
...
+	can_hold = list(/obj/item/ammo_magazine/standard_atgun)
+
+/obj/machinery/deployable/mounted/atgun/Initialize()
+	. = ..()
+	sponson = new sponson(src)
+
+/obj/machinery/deployable/mounted/atgun/attack_hand_alternate(mob/living/user)
+	return sponson.open(user)
+
+obj/item/storage/internal/ammo_rack/handle_mousedrop(mob/user, obj/over_object)
```
Comment (hyper2snyper) [inline code/modules/projectiles/guns/mounted.dm:266]:
```suggestion
/obj/item/storage/internal/ammo_rack/handle_mousedrop(mob/user, obj/over_object)
```

**Review by hyper2snyper — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/crates_lockers/largecrate_supplies.dm [hunk 17]
```diff
@@ -141,7 +141,15 @@
 	desc = "A supply crate containing two boxed TL-102 mounted heavy smartguns."
 	supplies = list(/obj/item/storage/box/tl102 = 2)
 
-
+/obj/structure/largecrate/supply/weapons/standard_atgun
+	name = "\improper TAT-36 anti tank gun and ammo chest (x1, x10)"
```
Comment (XSlayer300) [inline code/game/objects/structures/crates_lockers/largecrate_supplies.dm:145]:
```suggestion
	name = "\improper TAT-36 anti tank gun and ammo chest (x11)"
```
if the "sidearm weapons chest (x20)" contains "eight TP-44 revolvers, and twelve TP-14 service pistols" as two items, I thought i had to make this consistent.

File: code/modules/projectiles/ammo_datums.dm [hunk 18]
```diff
@@ -1322,6 +1322,45 @@ datum/ammo/bullet/revolver/tp44
...
+	damage = 100
+	penetration = 50
+	sundering = 10
+	handful_amount = 1
+
+/datum/ammo/rocket/atgun_shell/drop_nade(turf/T)
+	explosion(T, 0, 2, 3, 2)
+
+/datum/ammo/rocket/atgun_shell/apcr
+	name = "tungsten penetrator"
```
Comment (XSlayer300) [inline code/modules/projectiles/ammo_datums.dm:1341]:
```suggestion
	name = "tungsten penetrator shell"
```

**Comment by usnpeepoo** [comment]
beast PR with a weapon that instantly kills ancient runners from 14 tiles away

**Comment by Firefox13** [comment]
Apparently the gun:
cannot be targeted by ranged attacks (acid) making it much more durable than one could expect
can shoot through LZ shutters, probably gibbing an unwary xeno

**Comment by Pariah919** [comment]
>cannot be targeted by ranged attacks (acid) making it much more durable than one could expect
can shoot through LZ shutters, probably gibbing an unwary xeno
>
not sure about 1 but 2 is just a result of the gun flags, also a single shot can't gib you unless you're like, a young runner or sentinel

**Comment by jroinc** [comment]
Apparently the *explosion* appears to be IFF, and won't hurt marines if on aim mode.

Is that intended, or did I just see a janky shot?

**Comment by hyper2snyper** [comment]
> Apparently the _explosion_ appears to be IFF, and won't hurt marines if on aim mode.
> 
> Is that intended, or did I just see a janky shot?

explosion code doesnt have support for iff. so im assuming its a janky shot. Explosion code is janky anyways so might be something with that.

**Comment by Lefunny** [comment]
quite op weapon, gotta need nerfs after test merge

**Comment by Lumipharon** [comment]
Because you can't pick it up, so there is no way to cancel the acid. Normal weapons or deployables you just pick up, take a bit of burn damage and the acid stops.

**Comment by Losenis** [comment]
Just had it acided. If it's unacidable, it's bugged and still is.

**Comment by Shirbu** [comment]
APHE instakills ancient runners from exceptional range.

https://user-images.githubusercontent.com/23252965/156865344-c84cac71-220e-4fd9-a690-907765b1767b.mp4

**Comment by Losenis** [comment]
If you manage to hit a runner at that range, you probably deserve the kill.

Conversely, if you're a runner standing completely still in the open when we already have offscreen sadars and the like, it's probably a skill issue.

**Comment by BigBigga** [comment]
I hope this gun's merged for good; extremely based. I can't take this ''waaa waaa enough to gib runner waa'' seriously. Like those before me have said, if you're a runner, who stood perfectly still, in the open, long enough for the TAT to spot and shell you, you deserved it.

**Comment by Losenis** [comment]
And near a light source I might add, because at that range a good half of the screen is essentially not even rendered unless it's spammed with flares.

**Comment by usnpeepoo** [comment]
Maybe it wouldn’t be that bad if you could see it without getting shit on from a screen and a half away

**Pull request merged**